psi4/src/psi4/libmints/matrix.cc

```diff
@@ -2833,6 +2852,40 @@ void Matrix::write_to_dpdfile2(dpdfile2 *outFile) {
     global_dpd_->file2_mat_close(outFile);
 }
 
+void Matrix::write_to_dpdbuf4(dpdbuf4 *outBuf) {
+    if (outBuf->params->nirreps != nirrep_) {
+        char *str = new char[100];
+        sprintf(str, "Irrep count mismatch.  Matrix class has %d irreps, but dpdbuf4 has %d.", nirrep_,
+                outBuf->params->nirreps);
+        throw SanityCheckError(str, __FILE__, __LINE__);
```

Comment:
As a general statement, it might be good to switch to `stringstream` here to keep with C++11+ practices.